lib/services/elasticsearch.js

```diff
@@ -153,718 +168,635 @@ class ElasticSearch extends Service {
+   * Adds a mapping definition to a specific type
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} mappings - Collection mappings in ES format
+   *
+   * @return {Promise.<Object>} { }
    */
-  createCollection(request) {
-    const esRequest = initESRequest(request);
-
-    return this.kuzzle.indexCache.exists(esRequest.index)
-      .then(indexExists => {
-        if (! indexExists) {
-          this.throw('index_does_not_exist');
-        }
-
-        return this.kuzzle.indexCache.exists(esRequest.index, esRequest.type);
-      })
-      .then(collectionExists => {
-        if (collectionExists) {
-          return this.updateMapping(request);
-        }
-
-        const requestBody = request.input.body || {};
+  updateMapping (index, collection, mappings) {
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
 
-        this._checkMapping(requestBody);
+    return this.getMapping(index, collection, true)
+      .then(collectionMappings => {
+        this._checkMappings(collectionMappings);
 
-        esRequest.body = this._mergeDefaultMapping(
-          esRequest.index,
-          {},
-          requestBody
-        );
+        esRequest.body = _.merge(mappings, collectionMappings);
 
         return this.client.indices.putMapping(esRequest)
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
+          .then(() => {})
+          .catch(error => this.esWrapper.reject(error));
       });
   }
 
   /**
-   * Empty the content of a collection. Keep the existing mapping.
+   * Empties the content of a collection. Keep the existing mapping.
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   *
+   * @returns {Promise.<Object>} { }
    */
-  truncateCollection(request) {
-    const
-      deleteRequest = new Request({
-        index: request.input.resource.index,
-        collection: request.input.resource.collection,
-        body: {
-          query: {
-            match_all: {}
-          }
-        }
-      }, {
-        user: request.context.user
-      });
+  truncateCollection (index, collection) {
+    let mappings;
+
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
 
-    deleteRequest.input.args.refresh = request.input.args.refresh || false;
+    return this.getMapping(index, collection)
+      .then(collectionMappings => {
+        mappings = collectionMappings;
 
-    return this.deleteByQuery(deleteRequest)
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => this.client.indices.create({ ...esRequest, mappings }))
+      .then(() => {})
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this function is empty or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1186)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AEmptyBlock)